pkg/scaffold/helm/chart.go

```diff
@@ -15,40 +15,185 @@
 package helm
 
 import (
+	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/operator-framework/operator-sdk/pkg/scaffold"
 
+	"github.com/iancoleman/strcase"
 	log "github.com/sirupsen/logrus"
 	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/downloader"
+	"k8s.io/helm/pkg/getter"
+	"k8s.io/helm/pkg/helm/environment"
+	"k8s.io/helm/pkg/helm/helmpath"
 	"k8s.io/helm/pkg/proto/hapi/chart"
+	"k8s.io/helm/pkg/repo"
 )
 
-// HelmChartsDir is the relative directory within an SDK project where Helm
-// charts are stored.
-const HelmChartsDir string = "helm-charts"
+const (
 
-// CreateChartForResource creates a new helm chart in the SDK project for the
-// provided resource.
-func CreateChartForResource(r *scaffold.Resource, projectDir string) (*chart.Chart, error) {
-	log.Infof("Create %s/%s/", HelmChartsDir, r.LowerKind)
+	// HelmChartsDir is the relative directory within an SDK project where Helm
+	// charts are stored.
+	HelmChartsDir string = "helm-charts"
+
+	// DefaultAPIVersion is the Kubernetes CRD API Version used for fetched
+	// charts when the --api-version flag is not specified
+	DefaultAPIVersion string = "charts.helm.k8s.io/v1alpha1"
+)
+
+type FetchChartOptions struct {
+	Chart   string
+	Repo    string
+	Version string
+}
+
+func CreateChart(apiVersion, kind string, opts FetchChartOptions, projectDir string) (*scaffold.Resource, *chart.Chart, error) {
+	chartsDir := filepath.Join(projectDir, HelmChartsDir)
+	err := os.MkdirAll(chartsDir, 0755)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var (
+		r *scaffold.Resource
+		c *chart.Chart
+	)
+
+	// If we don't have a helm chart reference, scaffold the default chart
+	// from Helm's default template. Otherwise, fetch it.
+	if len(opts.Chart) == 0 {
+		r, c, err = scaffoldChart(apiVersion, kind, chartsDir)
+	} else {
+		r, c, err = fetchChart(apiVersion, kind, opts, chartsDir)
+	}
+	if err != nil {
+		return nil, nil, err
+	}
+	log.Infof("Create %s/%s/", HelmChartsDir, c.GetMetadata().GetName())
+	return r, c, nil
+}
+
+func scaffoldChart(apiVersion, kind, destDir string) (*scaffold.Resource, *chart.Chart, error) {
+	r, err := scaffold.NewResource(apiVersion, kind)
+	if err != nil {
+		return nil, nil, err
+	}
 
 	chartfile := &chart.Metadata{
-		Name:        r.LowerKind,
+		Name:        strcase.ToDelimited(kind, '-'),
```

Comment:
Just curious to know why this change.

Reply:
Many of the multi-word chart names use the hyphenated format: https://github.com/helm/charts/tree/master/stable

However, I'm looking at the [helm chart naming convention](https://github.com/helm/helm/blob/master/docs/chart_best_practices/conventions.md#chart-names), which says hyphens are allowed. But based on some of the problems mentioned in the [linked issue](https://github.com/helm/helm/issues/2192), it seems like it would be better to just avoid all of that. I'll revert back to `r.LowerKind`.